### DISCUSSION TITLE
Code block caption disappears

### DISCUSSION THREAD

**Discussion by blaplafla13th** [jgm/pandoc#9654] [Q&A] [answered]
When i remove --listings option and use --syntax-highlight, code block caption disappear but if i keep --listings color is gone but caption came back.

How to fix this?

**Comment by tarleb** [comment]
I'm not sure I understand. Could you provide an example and the pandoc command that you're running?

**Reply by blaplafla13th** [reply]
with this code block
```
\`\`\` { .python caption="hello world"}
print("hello")
\`\`\`
```

(sorry i dont know to write ``` in code block)

if i run `pandoc --pdf-engine xelatex -o test.pdf --highlight-style tango` i get code block rendered with color and no caption

if i run `pandoc --pdf-engine xelatex -o test.pdf --highlight-style tango --listings` i get code block rendered with no color but have caption

How can i get both color from highlight-style and caption?

**Reply by jgm** [reply]
You can't.  This treatment of `caption` is specific to listings, but listings doesn't do color.
I think there may be third-party filters that add captions to code blocks, though.

**Reply by blaplafla13th** [reply, marked as answer]
i will try write one by myself.
```
local pandoc=require('pandoc')
function Meta(meta)
    if PANDOC_WRITER_OPTIONS.listings and not FORMAT:match 'latex' then return nil end
  includes = [[
\usepackage{newfloat}
\DeclareFloatingEnvironment[
    fileext=los,
    name=Verbatim,
    placement=tbhp,
    within=section,
]{coding}
]]

  if meta['header-includes'] then
    table.insert(meta['header-includes'], pandoc.RawBlock('tex', includes))
  else
    meta['header-includes'] = List:new{pandoc.RawBlock('tex', includes)}
  end

  return meta
end

function CodeBlock(elem)
    if PANDOC_WRITER_OPTIONS.listings and not FORMAT:match 'latex' then return nil end
    if not elem.attributes['caption'] then return nil end
    return pandoc.RawBlock('tex','\\begin{coding}\n'
    .. pandoc.write ( pandoc.Pandoc({elem}),'latex' )
    .. '\\caption{'.. elem.attributes['caption'] ..'}\n'
    .. '\\end{coding}')
end

return {{Meta = Meta},{CodeBlock=CodeBlock}}
```
